lib/opencensus/trace/span_context.rb

```diff
@@ -0,0 +1,235 @@
+# Copyright 2017 OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module OpenCensus
+  module Trace
+    ##
+    # Span represents a span in a trace record. Spans are contained in
+    # a trace and arranged in a forest. That is, each span may be a root span
+    # or have a parent span, and may have zero or more children.
+    #
+    class SpanContext
+      ## @private Internal struct that holds trace-wide data.
+      TraceData = Struct.new :trace_id, :trace_options, :span_map, :rack_env
+      ## @private Internal struct that holds parsed trace context data.
+      TraceContext = Struct.new :trace_id, :span_id, :trace_options
+
+      class << self
+        ##
+        # Create a new root SpanContext object, given either a Trace-Context
+        # header value by itself, or an entire Rack environment. If a valid
+        # Trace-Context header can be obtained from either source, it is used
+        # to generate the SpanContext. Otherwise, a new root context with a
+        # unique `trace_id` and a root `span_id` of 0 is used.
+        #
+        # @param [String] header A Trace-Context header (optional)
+        # @param [Hash] rack_env The Rack environment hash (optional)
+        #
+        # @return [SpanContext]
+        #
+        def create_root header: nil, rack_env: nil
+          header ||= rack_env["HTTP_TRACE_CONTEXT"] if rack_env
+          trace_context = parse_trace_context_header header if header
```

Comment:
FYI, this format will be configurable as we'll need to support multiple formats

Reply:
Yes true. The struct is internal, and we'll deal with changes in the fields internally as needed.